spec/GetObjectKey.coffee

```diff
@@ -0,0 +1,186 @@
+noflo = require 'noflo'
+
+unless noflo.isBrowser()
+  chai = require 'chai' unless chai
+  GetObjectKey = require '../components/GetObjectKey.coffee'
+else
+  GetObjectKey = require 'noflo-objects/components/GetObjectKey.js'
+
+expect = chai.expect unless expect
+
+describe 'GetObjectKey', ->
+  c = null
+
+  beforeEach ->
+    c = GetObjectKey.getComponent()
+
+  describe 'inPorts', ->
+    it 'should include "in"', ->
+      expect(c.inPorts.in).to.be.an 'object'
+
+    it 'should include "key"', ->
+      expect(c.inPorts.key).to.be.an 'object'
+
+    it 'should include "sendgroup"', ->
+      expect(c.inPorts.sendgroup).to.be.an 'object'
+
+  describe 'outPorts', ->
+    it 'should include "out"', ->
+      expect(c.outPorts.out).to.be.an 'object'
+
+    it 'should include "object"', ->
+      expect(c.outPorts.object).to.be.an 'object'
+
+    it 'should include "missed"', ->
+      expect(c.outPorts.missed).to.be.an 'object'
+
+  describe 'data flow', ->
+    inIn = null
+    keyIn = null
+    sendgroupIn = null
+    outOut = null
+    objectOut = null
+    missedOut = null
+
+    beforeEach ->
+      inIn = noflo.internalSocket.createSocket()
+      keyIn = noflo.internalSocket.createSocket()
+      sendgroupIn = noflo.internalSocket.createSocket()
+      objectOut = noflo.internalSocket.createSocket()
+      missedOut = noflo.internalSocket.createSocket()
+      outOut = noflo.internalSocket.createSocket()
+
+      c.inPorts.in.attach inIn
+      c.inPorts.key.attach keyIn
+      c.inPorts.sendgroup.attach sendgroupIn
+      c.outPorts.out.attach outOut
+      c.outPorts.object.attach objectOut
+      c.outPorts.missed.attach missedOut
+
+    describe 'with input on all ports', ->
+      it 'should get the key', (done) ->
+        outOut.on 'data', (data) ->
+          chai.expect(data).to.eql 'canada'
+          done()
+
+        objectOut.on 'data', (data) ->
+          chai.expect(data).to.eql {test: true, eh: 'canada'}
+
+        keyIn.send 'eh'
+        inIn.send {test: true, eh: 'canada'}
+
+    describe 'when it has data that will miss', ->
+      it 'should trigger missed and not send object out as well', (done) ->
+        triggeredOut = false
+        triggeredMissed = false
+        outOut.on 'data', (data) ->
+          triggeredOut = true
+          if triggeredMissed and triggeredOut
```

Comment:
Since `triggeredOut` is always going to be true, isn't better to only check `triggeredMissed`?